core/loggerWithIdentifier.go

```diff
@@ -0,0 +1,88 @@
+package core
+
+import (
+    logger "github.com/ElrondNetwork/elrond-go-logger"
+)
+
+// loggerWithIdentifier is a decorator for the logger
+type loggerWithIdentifier struct {
+    logger     logger.Logger
+    identifier string
+}
+
+// NewLoggerWithIdentifier creates a new loggerWithIdentifier instance
+func NewLoggerWithIdentifier(logger logger.Logger, identifier string) *loggerWithIdentifier {
+    if logger == nil {
+        return nil
```

Comment:
interesting. We usually do not silent drop an error returning nil instance on a constructor. 
However, this will greatly aid code compaction when we will use this wrapper as we rely on the other constructors to check for nil the provided logger.